static/src/javascripts/projects/common/modules/experiments/tests/facebook-like-prompt.js

```diff
@@ -0,0 +1,176 @@
+            options = options || {};
+            var textTarget = (edition.textTarget || edition.target.toLocaleString());
+            var percentageIncomplete = 100 - percentageComplete;
+            var completeLikesText = likes.toLocaleString() + ' likes';
+            var incompleteLikesText = remainingLikes.toLocaleString() + ' to go';
+            return template(
+                '<p class="site-message__message" id="site-message__message">{{messageText}}</p>' +
+                '<div class="site-message--facebook-like-prompt__like ' +
+                             (options.showIncompleteLabelInsideProgressBar
+                                 ? 'site-message--facebook-like-prompt__like--inside-label'
+                                 : '') +
+                    '">' +
+                    '<div class="fb-like"' +
+                         'data-href="https://www.facebook.com/' + username + '"' +
+                         'data-layout="button"' +
+                         'data-action="like"' +
+                         'data-show-faces="true"' +
+                         'data-share="false"></div>' +
+                    '<div class="progress-bar" data-percentage="50%">' +
+                        '<div class="progress-bar__complete"' +
+                             'style="width: ' + percentageComplete + '%"' +
+                             'title="' + completeLikesText + '">' + completeLikesText + '</div>' +
+                        '<div class="progress-bar__incomplete"' +
+                              'style="width: ' + percentageIncomplete + '%"' +
+                              'title="' + incompleteLikesText + '">' +
+                              (options.showIncompleteLabelInsideProgressBar
+                                  ? incompleteLikesText
+                                  : '') +
+                        '</div>' +
+                    '</div>' +
+                    (!options.showIncompleteLabelInsideProgressBar
+                        ? '<div class="site-message--facebook-like-prompt__incomplete-text">' +
+                              '< ' + incompleteLikesText +
+                          '</div>'
+                        : '') +
+                '</div>',
+                {
+                    messageText: 'Help us reach ' + textTarget + ' people: ' +
+                                 '<span class="u-color-black">Like the Guardian on Facebook</span>'
+                }
+            );
+        };
+
+        var getLikes = function (username) {
+            return ajax({ url: 'https://graph.facebook.com/' + username + '?fields=likes' })
+                .then(function (response) {
+                    return response.likes;
+                });
+        };
+
+        var loadFacebookSdk = function () {
+            // Render like button
+            // SDK must be loaded/initialised after the like button is added
+
+            // TODO: Why do we have to initialise twice? Test on mobile + desktop!!
```

Comment:
The FB SDK might already be loaded and initialised by `FacebookAuthorizer`. I had trouble detecting this, so in the end I ended up initialising the SDK again. FB SDK gives us a warning about double initialising.

![image](https://cloud.githubusercontent.com/assets/921609/7257769/e5f6c2ec-e84f-11e4-98e4-04362f424a55.png)